examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart

```diff
@@ -290,14 +311,35 @@ class _DemoBottomAppBar extends StatelessWidget {
       );
     }
     rowContents.addAll(<Widget> [
-      new IconButton(
-        icon: const Icon(Icons.search),
-        onPressed: () {},
+      // Generally, an icon button does not need to be wrapped in a Semantics object.
+      // When the button has a null onPressed callback, it will be disabled by default.
+      //
+      // However, for the purposes of this demo, we don't want the button to appear disabled.
```

Comment:
This is kinda odd. Why should a screen reader user have a different experience of the UI then somebody who doesn't use a screen reader?

Reply:
Since this is a demo, these are dummy buttons, not disabled buttons.  

I decided to tell the screen reader that they are disabled because when I tried it out with a screen reader, I ended up confused.  I partially blindfolded myself and tried tapping those buttons.  When the reader told me I could double-tap to activate them, I did and I was confused when I couldn't find anything changed.

Since the primary purpose of this demo is to showcase normal Flutter code, I am OK with removing those semantics objects.

I'll let you make the final call, since we should apply any solution we come up with to all no-op buttons in the gallery.